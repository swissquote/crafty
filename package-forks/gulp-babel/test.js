const { Transform } = require("node:stream");
const { test } = require("node:test");
const { expect } = require("expect");
const path = require("path");
const Vinyl = require("vinyl");
const sourcemap = require("vinyl-sourcemap");
const babel = require(".");

test("should transpile with Babel", () => {
  return new Promise(done => {
    const stream = babel({
      plugins: ["@babel/transform-block-scoping"]
    });

    stream.on("data", file => {
      expect(/var foo/.test(file.contents.toString())).toBe(true);
      expect(file.relative).toBe("fixture.js");
    });

    stream.on("end", done);

    stream.write(
      new Vinyl({
        cwd: __dirname,
        base: path.join(__dirname, "fixture"),
        path: path.join(__dirname, "fixture/fixture.jsx"),
        contents: Buffer.from("let foo;")
      })
    );

    stream.end();
  });
});

function initSourceMap() {
  return new Transform({
    objectMode: true,
    transform(file, enc, callback) {
      sourcemap.add(file, (sourcemapErr, updatedFile) => {
        if (sourcemapErr) {
          callback(sourcemapErr);
          return;
        }

        callback(null, updatedFile);
      });
    }
  });
}

function writeSourceMap() {
  return new Transform({
    objectMode: true,
    transform(file, enc, callback) {
      var self = this;

      sourcemap.write(file, (sourcemapErr, updatedFile, sourcemapFile) => {
        if (sourcemapErr) {
          callback(sourcemapErr);
          return;
        }

        self.push(updatedFile);
        if (sourcemapFile) {
          self.push(sourcemapFile);
        }

        callback();
      });
    }
  });
}

test("should generate source maps", () => {
  return new Promise(done => {
    const init = initSourceMap();
    const write = writeSourceMap();

    init
      .pipe(
        babel({
          plugins: ["@babel/transform-arrow-functions"]
        })
      )
      .pipe(write);

    write.on("data", file => {
      expect(file.sourceMap.sources).toEqual(["fixture.es2015"]);
      expect(file.sourceMap.file).toBe("fixture.js");
      const contents = file.contents.toString();
      expect(/function/.test(contents)).toBe(true);
      expect(/sourceMappingURL/.test(contents)).toBe(true);
      done();
    });

    init.write(
      new Vinyl({
        cwd: __dirname,
        base: path.join(__dirname, "fixture"),
        path: path.join(__dirname, "fixture/fixture.es2015"),
        contents: Buffer.from("[].map(v => v + 1)"),
        sourceMap: ""
      })
    );

    init.end();
  });
});

test("should generate source maps for file in nested folder", () => {
  return new Promise(done => {
    const init = initSourceMap();
    const write = writeSourceMap();
    init
      .pipe(
        babel({
          plugins: ["@babel/transform-arrow-functions"]
        })
      )
      .pipe(write);

    write.on("data", file => {
      expect(file.sourceMap.sources).toEqual(["nested/fixture.es2015"]);
      expect(file.sourceMap.file).toBe("nested/fixture.js");
      const contents = file.contents.toString();
      expect(/function/.test(contents)).toBe(true);
      expect(/sourceMappingURL/.test(contents)).toBe(true);
      done();
    });

    init.write(
      new Vinyl({
        cwd: __dirname,
        base: path.join(__dirname, "fixture"),
        path: path.join(__dirname, "fixture/nested/fixture.es2015"),
        contents: Buffer.from("[].map(v => v + 1)"),
        sourceMap: ""
      })
    );

    init.end();
  });
});

test("should pass the result of transform().metadata in file.babel", () => {
  return new Promise(done => {
    const stream = babel({
      plugins: [
        {
          post(file) {
            file.metadata.test = "metadata";
          }
        }
      ]
    });

    stream.on("data", file => {
      expect(file.babel).toEqual({ test: "metadata" });
    });

    stream.on("end", done);

    stream.write(
      new Vinyl({
        cwd: __dirname,
        base: path.join(__dirname, "fixture"),
        path: path.join(__dirname, "fixture/fixture.js"),
        contents: Buffer.from("class MyClass {};")
      })
    );

    stream.end();
  });
});

test("should not rename ignored files", () => {
  return new Promise(done => {
    const stream = babel({
      ignore: [/fixture/]
    });

    const inputFile = {
      cwd: __dirname
    };

    inputFile.base = path.join(inputFile.cwd, "fixture");
    inputFile.basename = "fixture.jsx";
    inputFile.path = path.join(inputFile.base, inputFile.basename);
    inputFile.contents = Buffer.from(";");

    stream
      .on("data", file => {
        expect(file.relative).toBe(inputFile.basename);
      })
      .on("end", done)
      .end(new Vinyl(inputFile));
  });
});

test("should not rename files without an extension", () => {
  return new Promise(done => {
    const stream = babel();

    const inputFile = {
      cwd: __dirname
    };

    inputFile.base = path.join(inputFile.cwd, "bin");
    inputFile.basename = "app";
    inputFile.path = path.join(inputFile.base, inputFile.basename);
    inputFile.contents = Buffer.from(";");

    stream
      .on("data", file => {
        expect(file.relative).toBe(inputFile.basename);
      })
      .on("end", done)
      .end(new Vinyl(inputFile));
  });
});
