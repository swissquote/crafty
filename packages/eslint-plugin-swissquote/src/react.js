const { addMissingRules } = require("./utils");

// EcmaScript 6 specific configuration
module.exports = {
  settings: {
    "import/resolver": {
      node: {
        extensions: [".js", ".jsx", ".json"]
      }
    },
    react: {
      version: "detect"
    }
  },
  parserOptions: {
    ecmaFeatures: {
      jsx: true
    }
  },
  rules: {
    // Recommended rules override
    "@swissquote/swissquote/react/display-name": 0, // Disabled as it generates false positives
    "@swissquote/swissquote/react/jsx-no-duplicate-props": [
      "error",
      { ignoreCase: true }
    ],
    "@swissquote/swissquote/react/no-deprecated": "warn",
    "@swissquote/swissquote/react/prop-types": [
      "warn",
      { skipUndeclared: true }
    ],

    // Swissquote Rules
    // Disabled for now as it has false positives on inline handlers
    // TODO :: re-enable jsx-handler-names once https://github.com/yannickcr/eslint-plugin-react/issues/2832 is fixed
    "@swissquote/swissquote/react/jsx-handler-names": "off",
    "@swissquote/swissquote/react/jsx-pascal-case": "error",
    "@swissquote/swissquote/react/no-did-mount-set-state": "error",
    "@swissquote/swissquote/react/no-did-update-set-state": "error",
    "@swissquote/swissquote/react/no-redundant-should-component-update":
      "error",
    "@swissquote/swissquote/react/no-typos": "error",
    "@swissquote/swissquote/react/no-unused-state": "error",
    "@swissquote/swissquote/react/no-will-update-set-state": "error",
    "@swissquote/swissquote/react/prefer-es6-class": ["error", "always"],
    "@swissquote/swissquote/react/prefer-stateless-function": [
      "error",
      { ignorePureComponents: true }
    ],
    "@swissquote/swissquote/react/void-dom-elements-no-children": "error"
  }
};

// Add all recommended configurations from eslint-plugin-react
addMissingRules(
  require("eslint-plugin-react").configs.recommended.rules,
  module.exports.rules
);

// Add all recommended configurations from eslint-plugin-react-hooks
addMissingRules(
  require("eslint-plugin-react-hooks").configs.recommended.rules,
  module.exports.rules
);

// Disable all the rules from eslint-plugin-react that are handled by Prettier
addMissingRules(
  require("eslint-config-prettier/react").rules,
  module.exports.rules
);
