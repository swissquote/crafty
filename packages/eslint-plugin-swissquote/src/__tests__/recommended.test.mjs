import test from "node:test";
import { expect } from "expect";
import initSnapshot from "@onigoetz/ntr-expect-snapshot";

import { prepareESLint } from "../../test_utils.js";

initSnapshot(import.meta.url);

const lint = prepareESLint("recommended");

test("Warns on console.log", async t => {
  const result = await lint(`console.log("Yeah");\n`);

  expect(result.messages).toMatchSnapshot();
  expect(result.warningCount).toEqual(1);
  expect(result.errorCount).toEqual(0);
});

test("Uses sonar plugin", async t => {
  const result = await lint(
    `
/* global openWindow, closeWindow, moveWindowToTheBackground */

function changeWindow(param) {
  if (param === 1) {
    openWindow();
  } else if (param === 2) {
    closeWindow();
  } else if (param === 1) {
    // Noncompliant    ^
    moveWindowToTheBackground();
  }
}
`
  );

  expect(result.messages).toMatchSnapshot();
  expect(result.warningCount).toEqual(0);
  expect(result.errorCount).toEqual(1);
});

test("Works fine with ES6 code", async t => {
  const result = await lint(
    `
class SkinnedMesh extends THREE.Mesh {
  constructor(geometry, materials) {
    super(geometry, materials);

    this.idMatrix = SkinnedMesh.defaultMatrix();
    this.bones = [];
    this.boneMatrices = [];
    //...
  }
  update(camera) {
    //...
    super.update();
  }
  get boneCount() {
    return this.bones.length;
  }
  set matrixType(matrixType) {
    this.idMatrix = SkinnedMesh[matrixType]();
  }
  static defaultMatrix() {
    return new THREE.Matrix4();
  }
}

const { op: a, lhs: { op: b }, rhs: c } = getASTNode();

const someString = \`In JavaScript this is
    not legal.\`;

const obj = {
    // __proto__
    __proto__: theProtoObj,
    // Shorthand for ‘someString: someString’
    someString,
    // Methods
    toString() {
     // Super calls
     return \`d \${  super.toString()}\`;
    },
    // Computed (dynamic) property names
    [\`prop_\${(() => 42)()}\`]: 42
};

`
  );

  expect(result.messages).toMatchSnapshot();
  expect(result.warningCount).toEqual(0);
  expect(result.errorCount).toEqual(22);
});

test("jsx-no-duplicate-props: works with different props", async t => {
  const result = await lint(
    `
export default function SomeComponent() {
  return <div id="" className="" />;
}
`
  );

  expect(result.messages.length).toEqual(0);
  expect(result.warningCount).toEqual(0);
  expect(result.errorCount).toEqual(0);
});

test("jsx-no-duplicate-props: works fails with the same prop", async t => {
  const result = await lint(
    `
export default function SomeComponent() {
  return <div id="" id="" />;
}
`
  );

  expect(result.messages).toMatchSnapshot();
  expect(result.warningCount).toEqual(1);
  expect(result.errorCount).toEqual(0);
});

test("no-did-mount-set-state: fails with setState in componentDidMount", async t => {
  const alternateLint = prepareESLint("recommended", {
    settings: { react: { version: "16.0.0" } }
  });
  const result = await alternateLint(
    `
import React from "react";
import PropTypes from "prop-types";

export default class MyComponent extends React.Component {
  componentDidMount() {
    this.setState({
      name: this.props.name.toUpperCase()
    });
  }
  render() {
    return <div>Hello {this.state.name}</div>;
  }
}

MyComponent.propTypes = {
  name: PropTypes.string.isRequired
};
`
  );

  expect(result.messages).toMatchSnapshot();
  expect(result.warningCount).toEqual(0);
  expect(result.errorCount).toEqual(2);
});

test("Incorrect usage of hooks: fails with setState in componentDidMount", async t => {
  const result = await lint(
    `
import { useState, useEffect } from "react";

export default function MyComponent() {
  // 1. Use the name state variable
  const [name, setName] = useState("Mary");

  // 2. Use an effect for persisting the form
  if (name !== null) {
    useEffect(() => {
      localStorage.setItem("formData", name);
    });
  }

  // 3. Use the surname state variable
  const [surname, setSurname] = useState("Poppins");

  // 4. Use an effect for updating the title
  useEffect(() => {
    document.title = \`\${name} \${surname}\`;
  });

  return <div>{name}</div>;
}
`
  );

  expect(result.messages).toMatchSnapshot();
  expect(result.warningCount).toEqual(0);
  expect(result.errorCount).toEqual(3);
});
