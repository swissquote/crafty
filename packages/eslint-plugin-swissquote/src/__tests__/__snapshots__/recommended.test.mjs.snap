// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Incorrect usage of hooks: fails with setState in componentDidMount//0`] = `Array [
  Object {
    "column": 16,
    "endColumn": 23,
    "endLine": 5,
    "line": 5,
    "message": "'setName' is assigned a value but never used.",
    "messageId": "unusedVar",
    "nodeType": "Identifier",
    "ruleId": "no-unused-vars",
    "severity": 2,
    "suggestions": Array [
      Object {
        "data": Object {
          "varName": "setName",
        },
        "desc": "Remove unused variable 'setName'.",
        "fix": Object {
          "range": Array [
            135,
            144,
          ],
          "text": "",
        },
        "messageId": "removeVar",
      },
    ],
  },
  Object {
    "column": 5,
    "endColumn": 14,
    "endLine": 9,
    "line": 9,
    "message": "React Hook \\"useEffect\\" is called conditionally. React Hooks must be called in the exact same order in every component render.",
    "nodeType": "Identifier",
    "ruleId": "@swissquote/swissquote/react-hooks/rules-of-hooks",
    "severity": 2,
  },
  Object {
    "column": 19,
    "endColumn": 29,
    "endLine": 15,
    "line": 15,
    "message": "'setSurname' is assigned a value but never used.",
    "messageId": "unusedVar",
    "nodeType": "Identifier",
    "ruleId": "no-unused-vars",
    "severity": 2,
    "suggestions": Array [
      Object {
        "data": Object {
          "varName": "setSurname",
        },
        "desc": "Remove unused variable 'setSurname'.",
        "fix": Object {
          "range": Array [
            372,
            384,
          ],
          "text": "",
        },
        "messageId": "removeVar",
      },
    ],
  },
]`;

exports[`Uses sonar plugin//0`] = `Array [
  Object {
    "column": 10,
    "endColumn": 22,
    "endLine": 3,
    "line": 3,
    "message": "'changeWindow' is defined but never used.",
    "messageId": "unusedVar",
    "nodeType": "Identifier",
    "ruleId": "no-unused-vars",
    "severity": 2,
    "suggestions": Array [
      Object {
        "data": Object {
          "varName": "changeWindow",
        },
        "desc": "Remove unused variable 'changeWindow'.",
        "fix": Object {
          "range": Array [
            65,
            273,
          ],
          "text": "",
        },
        "messageId": "removeVar",
      },
    ],
  },
]`;

exports[`Warns on console.log//0`] = `Array [
  Object {
    "column": 1,
    "endColumn": 12,
    "endLine": 1,
    "line": 1,
    "message": "Unexpected console statement.",
    "messageId": "unexpected",
    "nodeType": "MemberExpression",
    "ruleId": "no-console",
    "severity": 1,
    "suggestions": Array [
      Object {
        "data": Object {
          "propertyName": "log",
        },
        "desc": "Remove the console.log().",
        "fix": Object {
          "range": Array [
            0,
            20,
          ],
          "text": "",
        },
        "messageId": "removeConsole",
      },
    ],
  },
]`;

exports[`Works fine with ES6 code//0`] = `Array [
  Object {
    "column": 7,
    "endColumn": 18,
    "endLine": 1,
    "line": 1,
    "message": "'SkinnedMesh' is defined but never used.",
    "messageId": "unusedVar",
    "nodeType": "Identifier",
    "ruleId": "no-unused-vars",
    "severity": 2,
    "suggestions": Array [
      Object {
        "data": Object {
          "varName": "SkinnedMesh",
        },
        "desc": "Remove unused variable 'SkinnedMesh'.",
        "fix": Object {
          "range": Array [
            0,
            474,
          ],
          "text": "",
        },
        "messageId": "removeVar",
      },
    ],
  },
  Object {
    "column": 27,
    "endColumn": 32,
    "endLine": 1,
    "line": 1,
    "message": "'THREE' is not defined.",
    "messageId": "undef",
    "nodeType": "Identifier",
    "ruleId": "no-undef",
    "severity": 2,
  },
  Object {
    "column": 16,
    "endColumn": 21,
    "endLine": 21,
    "line": 21,
    "message": "'THREE' is not defined.",
    "messageId": "undef",
    "nodeType": "Identifier",
    "ruleId": "no-undef",
    "severity": 2,
  },
  Object {
    "column": 8,
    "endColumn": 39,
    "endLine": 25,
    "fix": Object {
      "range": Array [
        483,
        514,
      ],
      "text": "
  op: a,
  lhs: { op: b },
  rhs: c
",
    },
    "line": 25,
    "message": "Replace \`·op:·a,·lhs:·{·op:·b·},·rhs:·c·\` with \`⏎··op:·a,⏎··lhs:·{·op:·b·},⏎··rhs:·c⏎\`",
    "messageId": "replace",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
  Object {
    "column": 13,
    "endColumn": 14,
    "endLine": 25,
    "line": 25,
    "message": "'a' is assigned a value but never used.",
    "messageId": "unusedVar",
    "nodeType": "Identifier",
    "ruleId": "no-unused-vars",
    "severity": 2,
    "suggestions": Array [
      Object {
        "data": Object {
          "varName": "a",
        },
        "desc": "Remove unused variable 'a'.",
        "fix": Object {
          "range": Array [
            484,
            490,
          ],
          "text": "",
        },
        "messageId": "removeVar",
      },
    ],
  },
  Object {
    "column": 27,
    "endColumn": 28,
    "endLine": 25,
    "line": 25,
    "message": "'b' is assigned a value but never used.",
    "messageId": "unusedVar",
    "nodeType": "Identifier",
    "ruleId": "no-unused-vars",
    "severity": 2,
    "suggestions": Array [
      Object {
        "data": Object {
          "varName": "b",
        },
        "desc": "Remove unused variable 'b'.",
        "fix": Object {
          "range": Array [
            489,
            505,
          ],
          "text": "",
        },
        "messageId": "removeVar",
      },
    ],
  },
  Object {
    "column": 37,
    "endColumn": 38,
    "endLine": 25,
    "line": 25,
    "message": "'c' is assigned a value but never used.",
    "messageId": "unusedVar",
    "nodeType": "Identifier",
    "ruleId": "no-unused-vars",
    "severity": 2,
    "suggestions": Array [
      Object {
        "data": Object {
          "varName": "c",
        },
        "desc": "Remove unused variable 'c'.",
        "fix": Object {
          "range": Array [
            505,
            513,
          ],
          "text": "",
        },
        "messageId": "removeVar",
      },
    ],
  },
  Object {
    "column": 43,
    "endColumn": 53,
    "endLine": 25,
    "line": 25,
    "message": "'getASTNode' is not defined.",
    "messageId": "undef",
    "nodeType": "Identifier",
    "ruleId": "no-undef",
    "severity": 2,
  },
  Object {
    "column": 7,
    "endColumn": 10,
    "endLine": 30,
    "line": 30,
    "message": "'obj' is assigned a value but never used.",
    "messageId": "unusedVar",
    "nodeType": "Identifier",
    "ruleId": "no-unused-vars",
    "severity": 2,
    "suggestions": Array [
      Object {
        "data": Object {
          "varName": "obj",
        },
        "desc": "Remove unused variable 'obj'.",
        "fix": Object {
          "range": Array [
            593,
            888,
          ],
          "text": "",
        },
        "messageId": "removeVar",
      },
    ],
  },
  Object {
    "column": 3,
    "endColumn": 5,
    "endLine": 31,
    "fix": Object {
      "range": Array [
        609,
        611,
      ],
      "text": "",
    },
    "line": 31,
    "message": "Delete \`··\`",
    "messageId": "delete",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 3,
    "endLine": 32,
    "fix": Object {
      "range": Array [
        624,
        626,
      ],
      "text": "",
    },
    "line": 32,
    "message": "Delete \`··\`",
    "messageId": "delete",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
  Object {
    "column": 16,
    "endColumn": 27,
    "endLine": 32,
    "line": 32,
    "message": "'theProtoObj' is not defined.",
    "messageId": "undef",
    "nodeType": "Identifier",
    "ruleId": "no-undef",
    "severity": 2,
  },
  Object {
    "column": 3,
    "endColumn": 5,
    "endLine": 33,
    "fix": Object {
      "range": Array [
        654,
        656,
      ],
      "text": "",
    },
    "line": 33,
    "message": "Delete \`··\`",
    "messageId": "delete",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 3,
    "endLine": 34,
    "fix": Object {
      "range": Array [
        698,
        700,
      ],
      "text": "",
    },
    "line": 34,
    "message": "Delete \`··\`",
    "messageId": "delete",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
  Object {
    "column": 3,
    "endColumn": 5,
    "endLine": 35,
    "fix": Object {
      "range": Array [
        716,
        718,
      ],
      "text": "",
    },
    "line": 35,
    "message": "Delete \`··\`",
    "messageId": "delete",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 3,
    "endLine": 36,
    "fix": Object {
      "range": Array [
        729,
        731,
      ],
      "text": "",
    },
    "line": 36,
    "message": "Delete \`··\`",
    "messageId": "delete",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
  Object {
    "column": 5,
    "endColumn": 6,
    "endLine": 37,
    "fix": Object {
      "range": Array [
        750,
        751,
      ],
      "text": "",
    },
    "line": 37,
    "message": "Delete \`·\`",
    "messageId": "delete",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
  Object {
    "column": 5,
    "endColumn": 20,
    "endLine": 38,
    "fix": Object {
      "range": Array [
        770,
        785,
      ],
      "text": "return \`d \${",
    },
    "line": 38,
    "message": "Replace \`·return·\`d·\${··\` with \`return·\`d·\${\`",
    "messageId": "replace",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
  Object {
    "column": 3,
    "endColumn": 5,
    "endLine": 39,
    "fix": Object {
      "range": Array [
        807,
        809,
      ],
      "text": "",
    },
    "line": 39,
    "message": "Delete \`··\`",
    "messageId": "delete",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 3,
    "endLine": 40,
    "fix": Object {
      "range": Array [
        812,
        814,
      ],
      "text": "",
    },
    "line": 40,
    "message": "Delete \`··\`",
    "messageId": "delete",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
  Object {
    "column": 3,
    "endColumn": 5,
    "endLine": 41,
    "fix": Object {
      "range": Array [
        855,
        857,
      ],
      "text": "",
    },
    "line": 41,
    "message": "Delete \`··\`",
    "messageId": "delete",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 1,
    "endLine": 44,
    "fix": Object {
      "range": Array [
        889,
        890,
      ],
      "text": "",
    },
    "line": 43,
    "message": "Delete \`⏎\`",
    "messageId": "delete",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
]`;

exports[`jsx-no-duplicate-props: works fails with the same prop//0`] = `Array [
  Object {
    "column": 21,
    "endColumn": 26,
    "endLine": 2,
    "line": 2,
    "message": "This JSX property is assigned multiple times.",
    "messageId": "noDuplicateJsxProps",
    "nodeType": "JSXAttribute",
    "ruleId": "@swissquote/swissquote/@eslint-react/no-duplicate-jsx-props",
    "severity": 1,
  },
]`;

exports[`no-did-mount-set-state: fails with setState in componentDidMount//0`] = `Array [
  Object {
    "column": 5,
    "endColumn": 7,
    "endLine": 8,
    "line": 6,
    "message": "Do not call \`this.setState\` in \`componentDidMount\` outside of functions, such as callbacks.",
    "messageId": "noSetStateInComponentDidMount",
    "nodeType": "CallExpression",
    "ruleId": "@swissquote/swissquote/@eslint-react/no-set-state-in-component-did-mount",
    "severity": 2,
  },
  Object {
    "column": 13,
    "endColumn": 22,
    "endLine": 15,
    "line": 15,
    "message": "[Deprecated] Use TypeScript or another type-checking solution instead.",
    "messageId": "noPropTypes",
    "nodeType": "Identifier",
    "ruleId": "@swissquote/swissquote/@eslint-react/no-prop-types",
    "severity": 2,
  },
]`;
